src/main/java/gregtech/integration/jei/recipe/GTRecipeWrapper.java

```diff
@@ -76,9 +78,28 @@ public void getIngredients(@Nonnull IIngredients ingredients) {
 
         // Outputs
         if (!recipe.getOutputs().isEmpty() || !recipe.getChancedOutputs().isEmpty()) {
-            List<ItemStack> recipeOutputs = recipe.getOutputs()
-                    .stream().map(ItemStack::copy).collect(Collectors.toList());
-            currentItemSlot += recipeOutputs.size();
+            List<ItemStack> recipeOutputs;
+            boolean outputNeedsReplacing = false;
+            try {
+                outputNeedsReplacing = recipe.getOutputs().get(0).hasTagCompound() && recipe.getOutputs().get(0).isItemEqual(MetaItems.TOOL_DATA_STICK.getStackForm(1));
+            } catch (Exception ignored) {}
```

Comment:
Why is this in a try catch? Additionally, instead of requiring the data stick item, check if the NBTCompound's key is equal to the constant I suggested.